packages/runtime/container-runtime/src/runningSummarizer.ts

```diff
@@ -68,12 +71,34 @@ export class RunningSummarizer implements IDisposable {
             summaryCollection,
             cancellationToken,
             stopSummarizerCallback,
-            options);
+            options,
+            runtime);
 
         await summarizer.waitStart();
 
-        // Run the heuristics after starting
+        // Update heuristic counts
+        // By the time we get here, there are potentially ops missing from the heuristic summary counts
+        // Examples of where this could happen:
+        // 1. Op is processed during the time that we are initiating the RunningSummarizer instance but before we
+        //    listen for the op events (will get missed by the handlers in the current workflow)
+        // 2. Op was sequenced after the last time we summarized (op sequence number > summarize ref sequence number)
+        const diff = runtime.deltaManager.lastSequenceNumber - (
+            heuristicData.lastSuccessfulSummary.refSequenceNumber
+            + heuristicData.numSystemOps
+            + heuristicData.numNonSystemOps);
+        if (diff > 0) {
```

Comment:
can diff be negative here?

Reply:
It's not possible based on my understanding of what these properties are tracking. In any case, if the number is negative, we probably can't meaningfully increase the counters.